### PR TITLE
data/lab: Drop caches before benchmarking mmap_copy and copy

### PR DESCRIPTION
**Pull request by andreia-oca**

### CODE REVIEW

File: content/chapters/data/lab/content/process-memory.md [hunk 1]
```diff
@@ -566,31 +567,24 @@ and let's build the two executable files:
...
-student@os:~/.../lab/support/copy$ ./read_write_copy
-time passed 24232 microseconds
-
-student@os:~/.../lab/support/copy$ ./read_write_copy
-time passed 25131 microseconds
+Benchmarking read_write_copy on in.dat
+time passed 42011 microseconds
 ```
 
 As you can see, there isn't a difference between the two approaches.
```
Comment (teodutu) [inline content/chapters/data/lab/content/process-memory.md:581]:
```suggestion
Run the script a few more times.
As you can see, there isn't much of a difference between the two approaches.
```

File: content/chapters/data/lab/content/process-memory.md [hunk 2]
```diff
@@ -566,31 +567,24 @@ and let's build the two executable files:
...
 Although we would have expected the use of multiple system calls to cause overhead, it's too little compared to the memory copying overhead.
 
+If you inspect `benchmark_cp.sh`, you will notice a weird-looking command `sh -c "sync; echo 3 > /proc/sys/vm/drop_caches"`.
+This is used to disable a memory optimization that the kernel does.
```
Comment (teodutu) [inline content/chapters/data/lab/content/process-memory.md:585]:
```suggestion
This is used to disable a memory optimization that the kernel does.
It's called "buffer cache" and it's a mechanism by which the kernel caches data blocks from recently accessed files in memory.
```
Repetitio mater studiorum est.

File: content/chapters/data/lab/content/process-memory.md [hunk 3]
```diff
@@ -566,31 +567,24 @@ and let's build the two executable files:
 student@os:~/.../lab/support/copy$ make
 ```
 
-and run them:
+Run the `benchmark_cp.sh` script:
```
Comment (teodutu) [inline content/chapters/data/lab/content/process-memory.md:570]: Also mention that students should generate the input files first. You can use this makefile from the I/O lab: https://github.com/open-education-hub/operating-systems/blob/8ffedab662d2c2d20a6b0d75efe95d0f8365e47f/content/chapters/io/lab/support/file-mappings/Makefile

**Review by teodutu — CHANGES_REQUESTED** [review]
I made some suggestions regarding the Markdown text.

File: content/chapters/data/lab/content/process-memory.md (hunk 3 above)
Comment (andreia-oca) [reply content/chapters/data/lab/content/process-memory.md:570]: The file is generated as the first step of this tutorial step with `./generate.sh` [1]

 [1] https://github.com/open-education-hub/operating-systems/pull/200/files/0801fedca2c02b8493748ff831faef5ffd039b45#diff-b84e4b0bb765638417bef0254489e3e46a817500e6ab65cf5e18751378d52726R561

Comment (teodutu) [reply content/chapters/data/lab/content/process-memory.md:570]: Oh, right. It should be fine. Sorry, I didn't see it.

**Review by teodutu — APPROVED** [review]
:+1:

**Pull request merged**